src/lib/dusk.rs

```diff
@@ -0,0 +1,247 @@
+    fn sub(self, other: Self) -> Self {
+        Self(self.0 - other.0)
+    }
+}
+
+/// Multiplication
+impl Mul for Dusk {
+    type Output = Self;
+    fn mul(self, other: Self) -> Self {
+        let a = dusk::from_dusk(self.0);
+        let b = dusk::from_dusk(other.0);
+        Self(dusk::dusk(a * b))
+    }
+}
+
+/// Division
+impl Div for Dusk {
+    type Output = Self;
+    fn div(self, other: Self) -> Self {
+        Self(dusk::dusk(self.0 as f64 / other.0 as f64))
+    }
+}
+
+/// Equality
+impl PartialEq for Dusk {
+    fn eq(&self, other: &Self) -> bool {
+        self.0 == other.0
+    }
+}
+impl PartialEq<Lux> for Dusk {
+    fn eq(&self, other: &Lux) -> bool {
+        self.0 == *other
+    }
+}
+impl PartialEq<f64> for Dusk {
+    fn eq(&self, other: &f64) -> bool {
+        self.0 == dusk::dusk(*other)
+    }
+}
+
+/// Comparison
+impl Ord for Dusk {
+    fn cmp(&self, other: &Self) -> Ordering {
+        self.0.cmp(&*other)
+    }
+}
+
+impl PartialOrd for Dusk {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        self.0.partial_cmp(&other.0)
+    }
+}
+impl PartialOrd<Lux> for Dusk {
+    fn partial_cmp(&self, other: &Lux) -> Option<Ordering> {
+        self.0.partial_cmp(other)
+    }
+}
+impl PartialOrd<f64> for Dusk {
+    fn partial_cmp(&self, other: &f64) -> Option<Ordering> {
+        self.0.partial_cmp(&dusk::dusk(*other))
+    }
+}
+
+/// Conversion ops
+/// Conveninent conversion of primitives to and from Dusk
+
+/// Floats are used directly as Dusk value
+impl From<f64> for Dusk {
+    fn from(val: f64) -> Self {
+        if val < 0.0 {
+            panic!("Dusk type does not support negative values");
+        }
+        Self(dusk::dusk(val))
+    }
+}
+
+// orphan
+
+impl From<Dusk> for f64 {
+    fn from(val: Dusk) -> f64 {
+        dusk::from_dusk(*val)
+    }
+}
+
+impl From<&Dusk> for f64 {
+    fn from(val: &Dusk) -> f64 {
+        (*val).into()
+    }
+}
+
+/// Lux represent Dusk in their underlying unit type
+impl From<Lux> for Dusk {
+    fn from(lux: Lux) -> Self {
+        Self(lux)
+    }
+}
+
+/// Strings are parsed as Dusk values (floats)
+impl FromStr for Dusk {
+    type Err = ParseFloatError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        f64::from_str(s).map(Dusk::from)
+    }
+}
+
+/*
+#[allow(clippy::from_over_into)]
+impl Into<f64> for Dusk {
+    fn into(self) -> f64 {
+        dusk::from_dusk(self.0)
+    }
+}
+*/
+
+/// Dusk derefs into its underlying Lux amount
+impl Deref for Dusk {
+    type Target = Lux;
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+/// Display
+/// Let the user print stuff
+
+impl fmt::Display for Dusk {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let v: f64 = self.into();
+        f64::fmt(&v, f)
+    }
+}
+
+impl fmt::Debug for Dusk {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let v: f64 = self.into();
+        write!(f, "{}", v)
+    }
+}
```

Comment:
You can just add `Debug` to the derive of `Dusk`, as well as you did for `Copy`, `Clone` and `Eq`, and remove this.